Assets/MRTK/SDK/Features/Input/Handlers/ObjectManipulator.cs

```diff
@@ -538,7 +560,10 @@ private void HandleTwoHandManipulationStarted()
             }
             if (twoHandedManipulationType.HasFlag(TransformFlags.Move))
             {
-                MixedRealityPose pointerPose = GetPointersPose();
+                // If near manipulation, a pure grabpoint centroid is used for
+                // the initial pointer pose; if far manipulation, a more complex
+                // look-rotation-based pointer pose is used.
+                MixedRealityPose pointerPose = IsNearManipulation() ? new MixedRealityPose(GetPointersGrabPoint()) : GetPointersPose();
```

Comment:
Although the rest of the class seems to mostly ignore this, it looks like `IsNearManipulation()` gets cached when manipulation starts:

https://github.com/microsoft/MixedRealityToolkit-Unity/blob/c7ce0c49a7ecac2364f0b3ecc25cda34882b7470/Assets/MRTK/SDK/Features/Input/Handlers/ObjectManipulator.cs#L647-L650

I'm not sure if we'd run into out of sync conditions if we switch to using that, but it feels more performant than iterating through all the pointers every time this is called. Any thoughts on the feasibility using this cached value from the start of a manipulation?

Reply:
That's a great point Kurtis.

I'm not sure we ever switch interaction modes (near vs far) halfway through an interaction. @julenka do you have insight on this?

Reply:
It's a good thing to think about, but in this case I'm not too worried. Thanks for looking at this carefully Kurtis!